lib/oedipus_lex.rb

```diff
@@ -62,9 +55,30 @@ def to_ruby state, predicates, exclusive
         when /^:/, "nil" then
           "  [:state, #{action}]"
         else
+          uses_text = true
           "  #{action} text"
         end
 
+      if uses_text or (action and action.include? 'text')
```

Comment:
The results are stored in the `text` variable, which is used in the action. If the action doesn't use `text` then we don't need to store the result.